sync.sh

```diff
@@ -2,9 +2,79 @@
 
 set -e
 
-# Read input arguments
+# Define Help function
+
+Help()
+{
+   # Display Help
+   echo "Tool to fill in belgian TST file from trade republic monthly report."
+   echo "This tool also creates a draft mail with the generated document and a qr code to pay the tax."
+   echo
+   echo "Syntax: sh sync.sh [-h|s|p|c] input_folder output_folder"
+   echo "options:"
+   echo "h     Display this help and exit."
+   echo "s     Set the signature image file path, default value is 'data/signature.jpg'."
+   echo "p     Set the personal informations json file path, default value is 'data/citizen.json'."
+   echo "c     Set the Gmail credentials json file path, default value is 'data/credentials.json'."
+   echo "arguments:"
+   echo "input_folder     Set the input folder for the pdf(s), default value is 'input'."
+   echo "output_folder    Set the output folder for the generated pdf(s) and qr code, default value is 'output'."
+   echo
+   echo "Please make sure to provide options before arguments."
+   echo
+}
+
+# Set default values for options
+SIGNATURE_FILE="$(pwd)/data/signature.jpg"
+PERSONAL_INFOS_FILE="$(pwd)/data/citizen.json"
+CREDENTIALS_FILE="$(pwd)/data/credentials.json"
+
+# Traverse arguments
+while getopts ":hs:p:c:" option; do
+   case $option in
+      h) # display Help
+        Help
+        exit;;
+      s) # Signature file
+        SIGNATURE_FILE="$(pwd)/$OPTARG";;
+      p) # Personal informations file
+        PERSONAL_INFOS_FILE="$(pwd)/$OPTARG";;
+      c) # Gmail credentials file
+        CREDENTIALS_FILE="$(pwd)/$OPTARG";;
+      \?) # Invalid option
+        echo "Error: Invalid option '-$OPTARG'"
+        echo "Run 'sh sync.sh -h' to get more information about existing options."
+        exit;;
+   esac
+done
+
+echo " ________  ______   ________       __    __  ________  ______  __         ______  ";
+echo "/        |/      \ /        |     /  |  /  |/        |/      |/  |       /      \ ";
+echo "\$\$\$\$\$\$\$\$//\$\$\$\$\$\$  |\$\$\$\$\$\$\$\$/      \$\$ |  \$\$ |\$\$\$\$\$\$\$\$/ \$\$\$\$\$\$/ \$\$ |      /\$\$\$\$\$\$  |";
+echo "   \$\$ |  \$\$ \__\$\$/    \$\$ | ______ \$\$ |  \$\$ |   \$\$ |     \$\$ |  \$\$ |      \$\$ \__\$\$/ ";
+echo "   \$\$ |  \$\$      \    \$\$ |/      |\$\$ |  \$\$ |   \$\$ |     \$\$ |  \$\$ |      \$\$      \ ";
+echo "   \$\$ |   \$\$\$\$\$\$  |   \$\$ |\$\$\$\$\$\$/ \$\$ |  \$\$ |   \$\$ |     \$\$ |  \$\$ |       \$\$\$\$\$\$  |";
+echo "   \$\$ |  /  \__\$\$ |   \$\$ |        \$\$ \__\$\$ |   \$\$ |    _\$\$ |_ \$\$ |_____ /  \__\$\$ |";
+echo "   \$\$ |  \$\$    \$\$/    \$\$ |        \$\$    \$\$/    \$\$ |   / \$\$   |\$\$       |\$\$    \$\$/ ";
+echo "   \$\$/    \$\$\$\$\$\$/     \$\$/          \$\$\$\$\$\$/     \$\$/    \$\$\$\$\$\$/ \$\$\$\$\$\$\$\$/  \$\$\$\$\$\$/  ";
+echo "                                                                                  ";
```

Comment:
This looks very cool, except if the terminal's width is less than 82, and unfortunately the default basically everywhere is 80.  Do you think there is a way to make it fit in there? Or is the text size fixed?

Reply:
I managed to make it fit in the default terminal size by remove 1 character from the "-" and the "L".  This was the changes that preserves that best the initial drawing. However resizing the terminal smaller than default size will still mess up the drawing.